consensus/src/consensus_provider.rs

```diff
@@ -48,6 +48,7 @@ pub fn start_consensus(
         node_config.consensus.mempool_executed_txn_timeout_ms,
     ));
 
+    AptosVM::set_concurrency_level_once(node_config.execution.concurrency_level as usize);
```

Comment:
Ahh, would it be possible to read a global variable from the config space instead? Ideally this would be set on load: https://github.com/aptos-labs/aptos-core/blob/main/config/src/config/mod.rs#L229
but we don't want to config knowing about AptosVM. I don't think it matters if AptosVM knows about the config though.

The current approach won't enable parallel execution for state sync, which is a bummer and will require additional wiring.

Reply:
Great point about statesync, I am still often missing the big picture context in our codebase. As per @zekun000 's suggestion, moved this to setup_environment when the node is spun up, so it should cover all nodes, without relying on consensus setup.